vobject/icalendar.py

```diff
@@ -445,11 +445,38 @@ def getrruleset(self, addRDate=False):
                     # a Ruby iCalendar library escapes semi-colons in rrules,
                     # so also remove any backslashes
                     value = line.value.replace('\\', '')
+                    # If dtstart has no time zone, `until` shouldn't have one.
+                    # Note: isinstance(dtstart, datetime.date) is not sufficient
+                    # to check for date only because isinstance(datetime.datetime,
+                    # datetime.date) == True.
+                    ignoretz = (
+                        isinstance(dtstart, datetime.date) and
+                        not isinstance(dtstart, datetime.datetime))
```

Comment:
The value of ``dtstart`` is either ``DATE`` or ``DATE-TIME``. ``isinstance(dtstart, datetime.date)`` is always ``True``.